src/atlas/projection/detail/ProjectionImpl.cc

```diff
@@ -53,39 +51,27 @@ struct DerivateBuilder : public ProjectionImpl::DerivateFactory {
 
 struct DerivateForwards final : ProjectionImpl::Derivate {
     using Derivate::Derivate;
-    PointLonLat d(PointXY P) const override {
-        PointLonLat A(xy2lonlat(P));
-        PointLonLat B(xy2lonlat(PointXY::add(P, H_)));
-        return PointLonLat::div(PointLonLat::sub(B, A), normH_);
-    }
+    PointLonLat d(PointXY P) const override { return (xy2lonlat(P + H_) - xy2lonlat(P)) * invnH_; }
 };
 
 struct DerivateBackwards final : ProjectionImpl::Derivate {
     using Derivate::Derivate;
-    PointLonLat d(PointXY P) const override {
-        PointLonLat A(xy2lonlat(PointXY::sub(P, H_)));
-        PointLonLat B(xy2lonlat(P));
-        return PointLonLat::div(PointLonLat::sub(B, A), normH_);
-    }
+    PointLonLat d(PointXY P) const override { return (xy2lonlat(P) - xy2lonlat(P - H_)) * invnH_; }
 };
 
 struct DerivateCentral final : ProjectionImpl::Derivate {
     DerivateCentral(const ProjectionImpl& p, PointXY A, PointXY B, double h, double refLongitude):
-        Derivate(p, A, B, h, refLongitude), H2_{PointXY::mul(H_, 0.5)} {}
+        Derivate(p, A, B, h, refLongitude), H2_{H_ * 0.5} {}
     const PointXY H2_;
-    PointLonLat d(PointXY P) const override {
-        PointLonLat A(xy2lonlat(PointXY::sub(P, H2_)));
-        PointLonLat B(xy2lonlat(PointXY::add(P, H2_)));
-        return PointLonLat::div(PointLonLat::sub(B, A), normH_);
-    }
+    PointLonLat d(PointXY P) const override { return (xy2lonlat(P + H2_) - xy2lonlat(P - H2_)) * invnH_; }
 };
 
 }  // namespace
 
 ProjectionImpl::Derivate::Derivate(const ProjectionImpl& p, PointXY A, PointXY B, double h, double refLongitude):
     projection_(p),
     H_{PointXY::mul(PointXY::normalize(PointXY::sub(B, A)), h)},
-    normH_(PointXY::norm(H_)),
+    invnH_(1. / PointXY::norm(H_)),
```

Comment:
The only place I can think of where FE_DIV_BY_ZERO could originate is here.

Reply:
I haven't been able to trigger the FPEs yet; what combination of platform/compilers do these FPEs trigger on?